File: library/src/main/java/org/fbme/lib/iec61499/declarations/SystemDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.Declaration;
import org.fbme.lib.common.RootElement;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface SystemDeclaration extends Declaration, RootElement {

    @NotNull List<ApplicationDeclaration> getApplications();

    @NotNull List<DeviceDeclaration> getDevices();

    @NotNull List<SegmentDeclaration> getSegments();

    @NotNull List<Link> getLinks();

    @NotNull List<Mapping> getMappings();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/SystemDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.Declaration
import org.fbme.lib.common.RootElement

interface SystemDeclaration : Declaration, RootElement {
    val applications: MutableList<ApplicationDeclaration>
    val devices: MutableList<DeviceDeclaration>
    val segments: MutableList<SegmentDeclaration>
    val links: MutableList<Link>
    val mappings: MutableList<Mapping>
}
